plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt

```diff
@@ -188,6 +198,33 @@ class ToolkitWebviewBrowser(val project: Project) : LoginBrowser(project, Toolki
     }
 
     override fun prepareBrowser(state: BrowserState) {
+        selectionSettings.clear()
+
+        if (!inspectExistingConnection(project)) {
+            // existing connections
+            val bearerCreds = ToolkitAuthManager.getInstance().listConnections()
+                .filterIsInstance<AwsBearerTokenConnection>()
+                .associate {
+                    it.id to BearerConnectionSelectionSettings(it) { conn ->
+                        if (conn.isSono()) {
+                            loginBuilderId(CODECATALYST_SCOPES)
+                        } else {
+                            // TODO: rewrite scope logic, it's short term solution only
+                            val scopes = if (conn.isSono()) {
+                                CODECATALYST_SCOPES
+                            } else {
+                                listOf(IDENTITY_CENTER_ROLE_ACCESS_SCOPE)
```

Comment:
Should we have CODECATALYST_SCOPES here too?

Reply:
yea let me ask in our chat to confirm this

Reply:
Mano I think I will keep it as is for now

Reply:
basically Codecatalyst has to be enabled by explicitly click getStarted node or via the setup auth page

Reply:
i just think of a case: if the idc account doesn't have Codecatalyst application access/enabled, their login will always fail